core/src/main/java/org/testcontainers/utility/DynamicPollInterval.java

```diff
@@ -0,0 +1,34 @@
+package org.testcontainers.utility;
+
+import org.awaitility.pollinterval.PollInterval;
+
+import java.time.Duration;
+import java.time.Instant;
+
+/**
+ * Awaitility {@link org.awaitility.pollinterval.PollInterval} that takes execution time into consideration,
+ * to allow a constant poll-interval, as opposed to Awaitility's default poll-delay behaviour.
+ *
+ * @deprecated For internal usage only.
+ */
+@Deprecated
+public class DynamicPollInterval implements PollInterval {
+
+    final Duration interval;
+    Instant lastTimestamp;
+
+    public DynamicPollInterval(Duration interval) {
```

Comment:
nit: WDYT about making the constructor private and using `DynamicPollInterval.of(duration)` and `DynamicPollInterval.ofMillis(100)`?